src/state.rs

```diff
@@ -30,7 +30,7 @@ impl LogMessage {
 pub struct ApplicationState {
     messages: Vec<LogMessage>,
     scroll_messages_view: usize,
-    input: String,
+    input: Vec<char>,
```

Comment:
I have a question about this. Why the solution can not work with a String? I thought that a String was a Vec<char> below (or transformable to a Vec<char>)

Reply:
I just chose Vec<char> because its a drop in replacement, so it was an easy change. We can make this a String again (and we probably should since its more perform ant) but insert needs to account for bytes length, so instead of self.cursor +=1 it should be something like self.cursor += char.utf8_len()